pyk/src/pyk/kast/inner.py

```diff
@@ -936,7 +936,13 @@ def flatten_label(label: str, kast: KInner) -> list[KInner]:
     -   Input: Cons operation to flatten.
     -   Output: Items of cons list.
     """
-    if type(kast) is KApply and kast.label.name == label:
-        items = (flatten_label(label, arg) for arg in kast.args)
-        return [c for cs in items for c in cs]
-    return [kast]
+    flattened_args = []
+    rest_of_args = [kast]  # Rest of arguments in reversed order
+    while rest_of_args:
+        current_arg = rest_of_args.pop()
+        if type(current_arg) is KApply and current_arg.label.name == label:
+            next_args = reversed(list(current_arg.args))
```

Comment:
This reversing can be saved by prepending to the list.

Reply:
Would it be worth using something like [deque objects](https://docs.python.org/3/library/collections.html#collections.deque)? My understanding is that if `rest_of_args` grows big this can be inefficient. But I'm fine with whatever solution seems fine to you!

Reply:
Never mind my comment. On a closer look, this is the right approach since `reversed` just produces an iterator.